test/src/test/java/org/corfudb/runtime/view/ViewsMaintenanceTest.java

```diff
@@ -0,0 +1,111 @@
+
+        final int numWrites = 100;
+
+        //TODO(Maithem): sync garbage from other clients
+        for (int x = 0; x < numWrites; x++) {
+            table.put(String.valueOf(x), String.valueOf(x));
+        }
+
+        long sizeOfTableAfterWrite = sizeOf.deepSizeOf(table);
+
+        MultiCheckpointWriter mcw = new MultiCheckpointWriter();
+        mcw.addMap(table);
+        Token trimMark = mcw.appendCheckpoints(rt, "cp1");
+        rt.getAddressSpaceView().prefixTrim(trimMark);
+        rt.getViewsMaintenance().setObjectGCPeriodInSeconds(Duration.ofMinutes(0).getSeconds());
+        rt.getViewsMaintenance().runRuntimeGC();
+        long sizeOfTableAfterGc = sizeOf.deepSizeOf(table);
+        final int numOfSetsStreamSets = 3;
+        assertThat(sizeOfTableAfterGc)
+                .isLessThan(sizeOfTableAfterWrite - (numWrites * Long.BYTES * numOfSetsStreamSets));
+        assertThat(rt.getAddressSpaceView().getReadCache().asMap()).isEmpty();
+        rt.shutdown();
+        assertThat(rt.getViewsMaintenance().isStarted()).isFalse();
+    }
+
+    @Test
+    public void runSyncObjectsTest() throws Exception {
+        CorfuRuntime rt1 = getDefaultRuntime();
+
+        String streamName = "streamA";
+
+        CorfuTable<String, String> table1Rt1 = rt1.getObjectsView()
+                .build()
+                .setType(CorfuTable.class)
+                .setStreamName(streamName)
+                .open();
+
+        CorfuRuntime rt2 = getRuntime(rt1.getLayoutView().getLayout()).connect();
+
+        final long objectRefreshPeriod = 1;
+        rt2.getViewsMaintenance().setObjectSyncDelayInSeconds(objectRefreshPeriod);
+        rt2.getViewsMaintenance().start();
+
+        CorfuTable<String, String> table1Rt2 = rt2.getObjectsView()
+                .build()
+                .setType(CorfuTable.class)
+                .setStreamName(streamName)
+                .open();
+
+        long rt2TableVersion = ((CorfuCompileProxy) ((ICorfuSMR) table1Rt2).getCorfuSMRProxy())
+                .getUnderlyingObject().getVersionUnsafe();
+        assertThat(rt2TableVersion).isEqualTo(Address.NON_ADDRESS);
+
+        // populate the first client's map
+        final int numWrites = 5;
+        for (int x = 0; x < numWrites; x++) {
+            table1Rt1.put(String.valueOf(x), String.valueOf(x));
+        }
+
+        // Notice that we don't access table1Rt2's map, but we expect it to be
+        // synced
+        final int numRefreshCycles = 3;
+        Sleep.sleepUninterruptibly(Duration.ofSeconds(objectRefreshPeriod * numRefreshCycles));
+        long rt2TableVersionAfterSync = ((CorfuCompileProxy) ((ICorfuSMR) table1Rt2).getCorfuSMRProxy())
+                .getUnderlyingObject().getVersionUnsafe();
+        // Verify that it was synced up to the tail
+        assertThat(rt2TableVersionAfterSync).isEqualTo(rt2.getSequencerView().query().getToken().getSequence());
```

Comment:
Though it is the same, to avoid potential errors on  rt2, wouldn't it be safer to compare to rt1?